TypeDBConsole.java

```diff
@@ -651,18 +656,43 @@ private void runQueryPrintAnswers(TypeDBTransaction tx, TypeQLQuery query) {
         if (query instanceof TypeQLDefine) {
             tx.query().define(query.asDefine()).get();
             printer.info("Concepts have been defined");
+            hasUncommittedChanges = true;
         } else if (query instanceof TypeQLUndefine) {
             tx.query().undefine(query.asUndefine()).get();
             printer.info("Concepts have been undefined");
+            hasUncommittedChanges = true;
         } else if (query instanceof TypeQLInsert) {
             Stream<ConceptMap> result = tx.query().insert(query.asInsert());
-            printCancellableResult(result, x -> printer.conceptMap(x, tx));
+            AtomicBoolean changed = new AtomicBoolean(false);
+            printCancellableResult(result, x -> {
+                changed.set(true);
+                printer.conceptMap(x, tx);
+            });
+            if (changed.get()) hasUncommittedChanges = true;
         } else if (query instanceof TypeQLDelete) {
-            tx.query().delete(query.asDelete()).get();
-            printer.info("Concepts have been deleted");
+            Stream<ConceptMap> result = tx.query().match(query.asDelete().match());
+            AtomicInteger answerCount = new AtomicInteger();
+            printCancellableResult(result, x -> {
+                answerCount.getAndIncrement();
+                printer.conceptMap(x, tx);
```

Comment:
I think this is odd - we've introduce an strong inconsistency with our protocol, rather than a minor one, by printing out all the maps that are being deleted. The user doesn't really have any interested in the concepts from which is deleted, so we shouldn't print them (they may not exist in a bit anyway!). Using this to count is ok though.

Reply:
I suppose that's fine. Seems odd to do all the work just to throw it away in the end, but given that it's not in line with the eventual solution we're aiming for, we probably shouldn't introduce unrealistic user expectations by printing them now.